File: app/src/androidTest/java/com/thoughtworks/pafsilva/testautomationworkshop/utils/ReadDummyServerResponse.java
package com.thoughtworks.pafsilva.testautomationworkshop.utils;

import android.support.test.InstrumentationRegistry;

import java.io.IOException;
import java.io.InputStream;
import java.util.Scanner;

public class ReadDummyServerResponse {

    public static String readJsonFile(String fileName) throws IOException {
        InputStream jsonFixtureFile = InstrumentationRegistry.getContext().getAssets().open("files/" + fileName);
        Scanner s = new Scanner(jsonFixtureFile).useDelimiter("\\A");
        String result = s.hasNext() ? s.next() : "";
        return result;
    }
}

File: app/src/androidTest/java/com/thoughtworks/pafsilva/testautomationworkshop/utils/ReadDummyServerResponse.kt
package com.thoughtworks.pafsilva.testautomationworkshop.utils

import androidx.test.InstrumentationRegistry

import java.io.IOException
import java.io.InputStream
import java.util.Scanner

object ReadDummyServerResponse {

    fun readJsonFile(fileName: String): String {
        val jsonFixtureFile = InstrumentationRegistry.getContext().assets.open("files/$fileName")
        val s = Scanner(jsonFixtureFile).useDelimiter("\\A")
        return if (s.hasNext()) s.next() else ""
    }
}